apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Interval.cs

```diff
@@ -0,0 +1,296 @@
+
+            if (this.Months != other.Months)
+            {
+                return false;
+            }
+
+            if (this.Days != other.Days)
+            {
+                return false;
+            }
+
+            if (this.Nanoseconds != other.Nanoseconds)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Gets the object hashcode
+        /// </summary>
+        public override int GetHashCode() => this.Days.GetHashCode() + this.Months.GetHashCode() + this.Nanoseconds.GetHashCode();
+
+        /// <summary>
+        /// Creates an interval based on Months, Days and Nanoseconds.
+        /// </summary>
+        public static Interval FromMonthsDaysNanos(int months, int days, BigInteger nanos) => new Interval(months, days, nanos);
+
+        /// <summary>
+        /// Creates an interval based on the Months given.
+        /// </summary>
+        public static Interval FromMonths(int months) => new Interval(months, 0, 0);
+
+        /// <summary>
+        /// Creates an Interval based on the Days given.
+        /// </summary>
+        public static Interval FromDays(int days) => new Interval(0, days, 0);
+
+        /// <summary>
+        /// Creates an Interval based on the seconds given.
+        /// </summary>
+        public static Interval FromSeconds(BigInteger seconds) => new Interval(0, 0, NanosecondsInASecond * seconds);
+
+        /// <summary>
+        /// Creates an Interval based on the milliseconds given.
+        /// </summary>
+        public static Interval FromMilliseconds(BigInteger milliseconds) => new Interval(0, 0, NanosecondsInAMillisecond * milliseconds);
+
+        /// <summary>
+        /// Creates an Interval based on the microseconds given.
+        /// </summary>
+        public static Interval FromMicroseconds(BigInteger microseconds) => new Interval(0, 0, NanosecondsInAMicrosecond * microseconds);
+
+        /// <summary>
+        /// Creates an Interval based on the nanoseconds given.
+        /// </summary>
+        public static Interval FromNanoseconds(BigInteger nanoseconds) => new Interval(0, 0, nanoseconds);
+
+
+        private static int YearsToMonths(string years) => int.Parse(years) * 12;
+
+        private static BigInteger HoursToNanoseconds(string hours) => BigInteger.Parse(hours) * NanosecondsInAnHour;
+
+        private static BigInteger MinutesToNanoseconds(string minutes) => BigInteger.Parse(minutes) * NanosecondsInAMinute;
+
+        private static BigInteger SecondsToNanoseconds(string seconds)
+        {
+            string[] splitNumber = seconds.Split('.');
+
+            if (splitNumber.Length > 2)
+            {
+                throw new FormatException("Invalid Format");
+            }
+
+            decimal nanoseconds = decimal.Parse(seconds) * NanosecondsInASecond;
+
+            return (BigInteger) nanoseconds;
+        }
+
+        private bool validateInt(string number)
```

Comment:
Not sure why you need this? Just parse the int.

But also, if you need it, just `return !number.Contains('.');`